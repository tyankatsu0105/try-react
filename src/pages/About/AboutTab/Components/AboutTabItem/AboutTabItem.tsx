import React, { FunctionComponent, useMemo } from 'react';
import { RouteComponentProps } from 'react-router-dom';
import styled from '@emotion/styled';

const Row = styled.div`
  display: flex;
  align-items: center;
  font-weight: bold;
  color: #fff;
  background-color: #333;

  & + & {
    border-top: 1px solid #eee;
  }
`;

const Dl = styled.dl`
  & + & {
    margin-top: 10px;
  }
`;
const Dt = styled.dt`
  padding: 10px;
`;
const Dd = styled.dd`
  padding: 10px;
`;

interface Props {
  id: number;
  name: string;
  age: number;
}

const data = [
  {
    id: 1,
    name: '太郎',
    age: 25,
  },
  {
    id: 2,
    name: '花子',
    age: 20,
  },
  {
    id: 3,
    name: 'ジュニア',
    age: 1,
  },
];

interface MatchParams {
  userId: string;
}

interface Props extends RouteComponentProps<MatchParams> {}

export const AboutTabItem: FunctionComponent<Props> = ({ match }) => {
  const filterdData = useMemo(() => {
    return data.find((item) => {
      return item.id.toString() === match.params.userId;
    });
  }, [match.params.userId]);

  return (
    <>
      {filterdData && (
        <Dl>
          <Row>
            <Dt>名前</Dt>
            <Dd>{filterdData.name}</Dd>
          </Row>
          <Row>
            <Dt>年齢</Dt>
            <Dd>{filterdData.age}</Dd>
          </Row>
        </Dl>
      )}
    </>
  );
};
