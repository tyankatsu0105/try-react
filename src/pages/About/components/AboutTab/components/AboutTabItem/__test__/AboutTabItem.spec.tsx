import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { AboutTabItem } from '../AboutTabItem';

const mock: any = jest.fn();
const mockMatchParams = {
  params: {
    userId: '1',
  },
};

const props = {
  match: {
    ...mockMatchParams,
    ...mock,
  },
  history: mock,
  location: mock,
};

describe('AboutTabItem', () => {
  it('render', () => {
    const { baseElement } = render(<AboutTabItem {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('text', () => {
    const { getByText } = render(<AboutTabItem {...props} />);
    expect(getByText('太郎')).toBeInTheDocument();
  });
});
