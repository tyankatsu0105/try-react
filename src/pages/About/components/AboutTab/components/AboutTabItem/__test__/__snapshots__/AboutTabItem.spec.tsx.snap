// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutTabItem render 1`] = `
<body>
  <div>
    <dl
      class="css-1kbqon"
    >
      <div
        class="css-1yydugk"
      >
        <dt
          class="css-1q43t41"
        >
          名前
        </dt>
        <dd
          class="css-1q43t41"
        >
          太郎
        </dd>
      </div>
      <div
        class="css-1yydugk"
      >
        <dt
          class="css-1q43t41"
        >
          年齢
        </dt>
        <dd
          class="css-1q43t41"
        >
          25
        </dd>
      </div>
    </dl>
  </div>
</body>
`;
